### PR TITLE
Resolve #407: switch explorer statistics to GraphQL interface

### LINKED ISSUE #407: Port Statistics to GraphQL
Displays such as https://www.toiletmap.org.uk/explorer/statistics currently depend on the rest endpoint defined here: https://github.com/neontribe/gbptm/blob/develop/packages/api/src/rest/statistics.js

We'd like to model this data in the schema and make it available through GraphQL.

I suggest that at the same time we try to tidy up the construction of the mongoose queries used here, and house them in methods on the models where they belong: https://github.com/neontribe/gbptm/blob/develop/packages/api/src/db/loo.js

### PR DESCRIPTION
**Pull request by jthistle**
This aims to eventually address #407. 

I'm opening this PR to allow code review and to share what I'm working on.

### CODE REVIEW

File: packages/api/src/graphql/typeDefs/index.js [hunk 1]
```diff
@@ -115,6 +115,18 @@ const typeDefs = gql`
     loo: Loo
   }
 
+  """
+  A container type for various statistical counters
+  """
+  type Counters {
```
Comment (marxian) [inline packages/api/src/graphql/typeDefs/index.js]: It'b be totally amazing to document these properties... If you can figure from the logic what they _really_ mean ;-)

**Comment by jthistle** [comment]
Right, I think this is pretty much done. All the numbers come through GraphQL now, and any mongo queries are attached to the mongoose schema.

**Comment by jthistle** [comment]
Search is now moved (apart from the queries for the list of areas, which is something else I could now GraphQL-ify I suppose).

File: packages/api/src/db/report.js [hunk 2]
```diff
@@ -38,7 +38,7 @@ const ReportSchema = new Schema(
         }
 
         // check that it references us as the previous report
-        if (!next.previous.equals(this._id)) {
+        if (!next.previous.equaReportSchemals(this._id)) {
```
Comment (jthistle) [inline packages/api/src/db/report.js]: note to self: revert this line

File: packages/api/src/graphql/resolvers/index.js [hunk 3]
```diff
@@ -27,19 +28,60 @@ const looInfoResolver = property => {
 const resolvers = {
   Query: {
     loo: (parent, args) => Loo.findById(args.id),
-    loos: async (parent, args) => {
+    loos: async (parent, args, context) => {
+      console.log('search: ', args);
```
Comment (jthistle) [inline packages/api/src/graphql/resolvers/index.js]: note to self: remove this console log

**Comment by jthistle** [comment]
Aha, thank you very much.

**Pull request merged**